server/test/models/ckb_sync/persist_test.rb

```diff
@@ -374,6 +374,43 @@ class PersistTest < ActiveSupport::TestCase
       end
     end
 
+    test ".save_block generated block's miner hash should be nil when lock args is invalid" do
+      VCR.use_cassette("blocks/10") do
+        SyncInfo.local_inauthentic_tip_block_number
+        node_block = CkbSync::Api.instance.get_block(DEFAULT_NODE_BLOCK_HASH).deep_stringify_keys
+        node_block["commit_transactions"].first["outputs"].first["lock"]["args"] = ["c30257c81dde7766fc98882ff1e9f8e95abbe79345982e12c6a849de90cbbad1"]
+
+        local_block = CkbSync::Persist.save_block(node_block, "inauthentic")
+
+        assert_nil local_block.miner_hash
+      end
+    end
+
+    test ".save_block generated block's miner hash should be nil when binary hash is empty" do
+      VCR.use_cassette("blocks/10") do
+        SyncInfo.local_inauthentic_tip_block_number
+        node_block = CkbSync::Api.instance.get_block(DEFAULT_NODE_BLOCK_HASH).deep_stringify_keys
+        node_block["commit_transactions"].first["outputs"].first["lock"]["args"] = ["c30257c81dde7766fc98882ff1e9f8e95abbe79345982e12c6a849de90cbbad1"]
+
+        local_block = CkbSync::Persist.save_block(node_block, "inauthentic")
+
+        assert_nil local_block.miner_hash
+      end
+    end
+
+    test ".save_block generated block's miner hash should be nil when binary hash is invalid" do
+      VCR.use_cassette("blocks/10") do
+        SyncInfo.local_inauthentic_tip_block_number
+        node_block = CkbSync::Api.instance.get_block(DEFAULT_NODE_BLOCK_HASH).deep_stringify_keys
+        node_block["commit_transactions"].first["outputs"].first["lock"]["args"] = ["0xc30257c81dde7766fc98882ff1e9f8e95abbe79345982e12c6a849de90cbbad1"]
+        node_block["commit_transactions"].first["outputs"].first["lock"]["binary_hash"] = "0xc30257c81dde7766fc98882ff1e9f8e95abbe79345982e12c6a849de90cbbad1"
```

Comment:
Extract `node_block["commit_transactions"].first["outputs"].first["lock"]` maybe better.

Reply:
ok, I will fix this.